pandas/core/internals/blocks.py

```diff
@@ -811,17 +821,24 @@ def _replace_coerce(
             if value is None:
                 # gh-45601, gh-45836, gh-46634
                 if mask.any():
-                    nb = self.astype(np.dtype(object), copy=False)
-                    if nb is self and not inplace:
+                    has_ref = self.refs.has_reference()
+                    nb = self.astype(np.dtype(object), copy=False, using_cow=using_cow)
```

Comment:
``copy=False`` doesn't do anything here right since astyping will always make a copy?
https://github.com/pandas-dev/pandas/blob/29140d44f9e96197fc5c74f460e6b9d87f731ea9/pandas/core/dtypes/astype.py#L136-L138

Reply:
Haven't looked into this too closely, but I guess this would be a view when going from object -> object.

Happy to let optimizing this be a follow up.

Reply:
Yeah this is the case where I was able to produce the bug. Run time wise this is more or less optimal right now, the actual checks are performed within astype anyway

Reply:
I was referring to the ``nb.copy()`` below. I'm a little worried that that might double copy if astype already copied.

Reply:
Yeah this can still happen, but Astype should take care of returning a block without references if a copy already happened (check is not perfect yet though)

Reply:
Thanks for clarifying.